src/overlays/actors/ovl_En_Anubice/z_en_anubice.c

```diff
@@ -292,52 +299,53 @@ void EnAnubice_SetupDie(EnAnubice* this, GlobalContext* globalCtx) {
     this->animLastFrame = lastFrame;
     Animation_Change(&this->skelAnime, &gAnubiceFallDownAnim, 1.0f, 0.0f, lastFrame, ANIMMODE_ONCE, -20.0f);
 
-    this->unk_256 = false;
-    this->unk_258 = 0;
+    this->isNearWall = false;
+    this->fallTargetYaw = 0;
     this->deathTimer = 20;
     this->actor.velocity.x = this->actor.velocity.z = 0.0f;
     this->actor.gravity = -1.0f;
 
-    if (BgCheck_SphVsFirstPoly(&globalCtx->colCtx, &this->fireballPos, 70.0f)) {
-        this->unk_256 = true;
-        this->unk_258 = this->actor.shape.rot.x - 0x7F00;
+    if (BgCheck_SphVsFirstPoly(&globalCtx->colCtx, &this->headPos, 70.0f)) {
+        this->isNearWall = true;
+        this->fallTargetYaw = this->actor.shape.rot.x - 0x7F00;
     }
 
     this->actionFunc = EnAnubice_Die;
 }
 
 void EnAnubice_Die(EnAnubice* this, GlobalContext* globalCtx) {
     f32 curFrame;
-    f32 phi_f2;
-    Vec3f sp4C = { 0.0f, 0.0f, 0.0f };
+    f32 rotX;
+    Vec3f fireEffectInitialPos = { 0.0f, 0.0f, 0.0f };
     Vec3f fireEffectPos = { 0.0f, 0.0f, 0.0f };
     s32 pad;
 
     SkelAnime_Update(&this->skelAnime);
     Math_ApproachZeroF(&this->actor.shape.shadowScale, 0.4f, 0.25f);
 
-    if (this->unk_256) {
-        Math_SmoothStepToS(&this->actor.shape.rot.y, this->unk_258, 1, 10000, 0);
-        if (fabsf(this->actor.shape.rot.y - this->unk_258) < 100.0f) {
-            this->unk_256 = false;
+    // If near a wall, turn to avoid the tail falling through it during the death animation
+    if (this->isNearWall) {
+        Math_SmoothStepToS(&this->actor.shape.rot.y, this->fallTargetYaw, 1, 10000, 0);
+        if (fabsf(this->actor.shape.rot.y - this->fallTargetYaw) < 100.0f) {
+            this->isNearWall = false;
         }
     }
 
     curFrame = this->skelAnime.curFrame;
-    phi_f2 = curFrame * -3000.0f;
-    phi_f2 = CLAMP_MIN(phi_f2, -11000.0f);
+    rotX = curFrame * -3000.0f;
+    rotX = CLAMP_MIN(rotX, -11000.0f);
 
     Matrix_RotateY(BINANG_TO_RAD(this->actor.shape.rot.y), MTXMODE_NEW);
-    Matrix_RotateX(BINANG_TO_RAD(phi_f2), MTXMODE_APPLY);
-    sp4C.y = Rand_CenteredFloat(10.0f) + 30.0f;
-    Matrix_MultVec3f(&sp4C, &fireEffectPos);
+    Matrix_RotateX(BINANG_TO_RAD(rotX), MTXMODE_APPLY);
+    fireEffectInitialPos.y = Rand_CenteredFloat(10.0f) + 30.0f;
+    Matrix_MultVec3f(&fireEffectInitialPos, &fireEffectPos);
```

Comment:
the source vector is the model space position of the effect, roughly (assuming the rotX matches the death animation)
```suggestion
    Matrix_MultVec3f(&fireEffectModelPos, &fireEffectPos);
```
at this point `fireEffectPos` is the offset from the actor in world coordinates, and on the next lines becomes a world position, so I guess it can just stay "pos" as is 🤷